src/pip/_internal/utils/misc.py

```diff
@@ -542,10 +542,9 @@ def __str__(self) -> str:
         return self.redacted
 
     # This is useful for testing.
-    def __eq__(self, other: Any) -> bool:
-        if type(self) is not type(other):
+    def __eq__(self, other: object) -> bool:
+        if type(self) is not type(other) or not isinstance(other, HiddenText):
```

Comment:
This one is really about type narrowing, since mypy doesn't do that for `type(self) is not type(other)`. The reason for the change at all comes down to the fact that ``self.secret == other.secret`` isn't recognised as a bool when `other` is `Any`.